### PR TITLE
Feature/recommendations

### PR DESCRIPTION
**Pull request by korgon**
* Added `template` command with various sub-commands
* Added `context` used throughout to provide contextual information to scripts about the project that Snapfu is running in (repo info, user info, project info)
* Added check of Snapfu version to inform users when new version is available
* Help improvements and new `about` command

Additionally updated docs, packages and some tests. Also added commitizen, husky standard-version and lint-staged packages to use similar process to other Snap repos.

In order to fully test changes using '--dev' you must have springbox(bet/snap-recommendations branch) and smc-config-api(templatestubs branch)  running locally.

### CODE REVIEW

**Review by dkonieczek — DISMISSED** [review]

**Review by codeallthethingz — DISMISSED** [review]

**Comment by korgon** [comment]
This finally is ready for release! Just need some testing from another person to verify functionality and command flow.

File: src/init.js [hunk 1]
```diff
@@ -1,149 +1,290 @@
...
+				validate: (input) => {
+					return input && input.length > 0 && /^[0-9a-zA-Z]{32}$/.test(input);
+				},
+			},
 		];
+
 		const answers = await inquirer.prompt(questions);
-		if (config.dev) {
-			console.log(chalk.blueBright('dev mode skipping new repo creation'));
+		console.log();
```
Comment (dkonieczek) [inline src/init.js:102]: 👀

File: src/init.js [hunk 2]
```diff
@@ -1,149 +1,290 @@
...
+				'snapfu.framework': answers.framework,
+			});
+			console.log(`${chalk.greenBright(templateUrlHTTP)}\n`);
+		}
+
+		// save secretKey mapping to creds.json
+		const { siteId, secretKey } = await auth.saveSecretKey(answers.secretKey, answers.siteId);
+
+		await setRepoSecret(options, { siteId: answers.siteId, secretKey: answers.secretKey, organization: answers.organization, name: answers.name });
+		console.log();
```
Comment (dkonieczek) [inline src/init.js:181]: 👀

File: src/recs.js [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+		} else {
+			templates.forEach((template, index) => {
+				console.log(`${chalk.green(template.details.name)} ${branch ? `[${branch}]` : ''} ${chalk.blueBright(`(${template.path})`)}`);
+			});
+		}
+	}
+
+	if (!location || location == 'remote') {
+		if (!location) {
+			console.log();
```
Comment (dkonieczek) [inline src/recs.js:115]: 👀

File: src/secret.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+	const answers = await inquirer.prompt({
+		type: 'input',
+		name: 'secretKey',
+		message: 'Please enter the secretKey as found in the SMC console (32 characters):',
+		validate: (input) => {
+			return input && input.length > 0 && /^[0-9a-zA-Z]{32}$/.test(input);
+		},
+	});
+	console.log();
```
Comment (dkonieczek) [inline src/secret.js:22]: 👀

**Review by dkonieczek — APPROVED** [review]

**Pull request merged**